atm.cpp

```diff
@@ -0,0 +1,101 @@
+#include "atm.h"
+#include "server_access_layer.h"
+
+Atm::Atm(std::istream& in, std::ostream& out): _ui(in, out), _validator(), _serverAccessLayer(new ServerAccessLayer())
+{
+	_address._country = "TestCountry";
+	_address._city = "TestCity";
+	_address._street = "TestStreet";
+	_address._homeNumber = 1;
+}
+
+Atm::~Atm() {}
+
+void Atm::run() const
+{
+	_ui.greeting();
+	_getch();
+	_ui.clear();
+	std::string number = readCardNumber();
+	std::string pins = readPin();
+	DigitSequence<16> dscn(number);
+	DigitSequence<4> dspin(pins);
+	CardNumber cn(dscn);
+	Pin pin(dspin);
```

Comment:
```suggestion
	Pin pin(readPin());
```
Remove lines 19 - 22